components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/ImportUtils.java

```diff
@@ -385,7 +396,32 @@ public static API importApi(String extractedFolderPath, APIDTO importedApiDTO, B
     }
 
     /**
-     * This method is used to validate the Gateway environments from the deplotment enviornments file. Gateway
+     * Check whether an advertise only API
+     *
+     * @param importedApiDTO API DTO to import
+     */
+    public static boolean isAdvertiseOnlyAPI(APIDTO importedApiDTO) {
+        return importedApiDTO.getAdvertiseInfo() != null && importedApiDTO.getAdvertiseInfo().isAdvertised();
+    }
+    
+    /**
+     * Process the properties specific to advertise only APIs
+     *
+     * @param importedApiDTO               API DTO to import
+     * @param tokenScopes Scopes of the token
+     */
+    private static void processAdvertiseOnlyPropertiesInDTO(APIDTO importedApiDTO, String[] tokenScopes) {
```

Comment:
What is the reason behind the logic of this method?

Reply:
The main aim of this function to processAdvertiseOnly properties in the API DTO. Currently, this function just does one thing. (We can enhance this function in the future if we need)
As per the discussion with @malinthaprasan and @tharindu1st , we decided to allow only the users with admin privileges (who have apim:admin scope) to set the originalDevPortalUrl. Otherwise, the other users can inject malicious URLs to do attacks and etc.

Reply:
In that case I think the message should be a `log.info`, since its technically not an error.

Its also a good idea to add the above explanation in a comment above the if block.

Reply:
+1, Will do the changes accordingly.

Reply:
Fixed via https://github.com/wso2/carbon-apimgt/pull/10141/commits/0d054ff2d50f622137dfdd44e5b0ba035f8d8f26

Reply:
Shall we just make this a debug log. Since this is security related, we might not need an info log even. We can silently ignore these URLs.

Reply:
Fixed via https://github.com/wso2/carbon-apimgt/pull/10141/commits/0d054ff2d50f622137dfdd44e5b0ba035f8d8f26